riot_auth/auth.py

```diff
@@ -158,9 +158,68 @@ def __set_tokens_from_uri(self, data: Dict) -> None:
         data = dict(parse_qsl(result))
         self.__update(extract_jwt=True, **data)
 
+    async def __fetch_access_token(
+        self, session: aiohttp.ClientSession, body: Dict, headers: Dict, resp_type: str
+    ) -> bool:
+        multifactor_status = False
+
+        if resp_type != "response":  # not reauth
+            async with session.put(
+                "https://auth.riotgames.com/api/v1/authorization",
+                json=body,
+                headers=headers,
+            ) as r:
+                data: Dict = await r.json()
+                resp_type = data["type"]
+                if resp_type == "response":
+                    ...
+                elif resp_type == "auth":
+                    err = data.get("error")
+                    if err == "auth_failure":
+                        raise RiotAuthenticationError(
+                            f"Failed to authenticate. Make sure username and password are correct. `{err}`."
+                        )
+                    elif err == "rate_limited":
+                        raise RiotRatelimitError()
+                    else:
+                        raise RiotUnknownErrorTypeError(
+                            f"Got unknown error `{err}` during authentication."
+                        )
+                elif resp_type == "multifactor":
+                    multifactor_status = True
+                else:
+                    raise RiotUnknownResponseTypeError(
+                        f"Got unknown response type `{resp_type}` during authentication."
+                    )
+
+        self._cookie_jar = session.cookie_jar
+
+        if multifactor_status == False:
+            self.__set_tokens_from_uri(data)
```

Comment:
This will crash the program as `data` might not exist here due to https://github.com/floxay/python-riot-auth/pull/15/files#diff-ab6bc481930e7dd83e1867177417726e038e82578cbdb8768824e2943c2104f9R166